src/main/java/com/example/spring/service/user/UserServiceImpl.java

```diff
@@ -34,6 +37,10 @@ public UserResponseDto register(
                 .orElseThrow(() -> new EntityNotFoundException("Role USER not found"));
         user.setRoles(Set.of(userRole));
         user.setPassword(passwordEncoder.encode(request.getPassword()));
+        userRepository.save(user);
+        ShoppingCart shoppingCart = new ShoppingCart();

```

Comment:
It's better to create separate method in ShopCartService 
`cartService.createShoppingCart(user);`


```
public void createShoppingCart(User user) {
        ShoppingCart shoppingCart = new ShoppingCart();
        shoppingCart.setUser(user);
        repository.save(shoppingCart);
    }
```